bridges/BakaUpdatesMangaReleasesBridge.php

```diff
@@ -3,82 +3,154 @@ class BakaUpdatesMangaReleasesBridge extends BridgeAbstract {
-		if (!$cols)
-			returnServerError('No releases');
+			// content is an unstructured pile of divs, ugly to parse
+			$cols = $html->find('div#main_content div.row > div.text');
+			if (!$cols)
+				returnServerError('No releases');
 
-		$rows = array_slice(
-			array_chunk($cols, self::LIMIT_COLS), 0, self::LIMIT_ITEMS
-		);
+			$rows = array_slice(
+				array_chunk($cols, self::LIMIT_COLS), 0, self::LIMIT_ITEMS
+			);
 
-		if (isset($rows[0][1])) {
-			$this->feedName = $this->filterHTML($rows[0][1]->plaintext);
-		}
+			if (isset($rows[0][1])) {
+				$this->feedName = $this->filterHTML($rows[0][1]->plaintext);
+			}
 
-		foreach($rows as $cols) {
-			if (count($cols) < self::LIMIT_COLS) continue;
+			foreach($rows as $cols) {
+				if (count($cols) < self::LIMIT_COLS) continue;
 
-			$item = array();
-			$title = array();
+				$item = array();
+				$title = array();
 
-			$item['content'] = '';
+				$item['content'] = '';
 
-			$objDate = $cols[0];
-			if ($objDate)
-				$item['timestamp'] = strtotime($objDate->plaintext);
+				$objDate = $cols[0];
+				if ($objDate)
+					$item['timestamp'] = strtotime($objDate->plaintext);
 
-			$objTitle = $cols[1];
-			if ($objTitle) {
-				$title[] = $this->filterHTML($objTitle->plaintext);
-				$item['content'] .= '<p>Series: ' . $this->filterText($objTitle->innertext) . '</p>';
-			}
+				$objTitle = $cols[1];
+				if ($objTitle) {
+					$title[] = $this->filterHTML($objTitle->plaintext);
+					$item['content'] .= '<p>Series: ' . $this->filterText($objTitle->innertext) . '</p>';
+				}
+
+				$objVolume = $cols[2];
+				if ($objVolume && !empty($objVolume->plaintext))
+					$title[] = 'Vol.' . $objVolume->plaintext;
 
-			$objVolume = $cols[2];
-			if ($objVolume && !empty($objVolume->plaintext))
-				$title[] = 'Vol.' . $objVolume->plaintext;
+				$objChapter = $cols[3];
+				if ($objChapter && !empty($objChapter->plaintext))
+					$title[] = 'Chp.' . $objChapter->plaintext;
 
-			$objChapter = $cols[3];
-			if ($objChapter && !empty($objChapter->plaintext))
-				$title[] = 'Chp.' . $objChapter->plaintext;
+				$objAuthor = $cols[4];
+				if ($objAuthor && !empty($objAuthor->plaintext)) {
+					$item['author'] = $this->filterHTML($objAuthor->plaintext);
+					$item['content'] .= '<p>Groups: ' . $this->filterText($objAuthor->innertext) . '</p>';
+				}
 
-			$objAuthor = $cols[4];
-			if ($objAuthor && !empty($objAuthor->plaintext)) {
-				$item['author'] = $this->filterHTML($objAuthor->plaintext);
-				$item['content'] .= '<p>Groups: ' . $this->filterText($objAuthor->innertext) . '</p>';
+				$item['title'] = implode(' ', $title);
+				$item['uri'] = $this->getURI();
+				$item['uid'] = $this->getSanitizedHash($item['title']);
+
+				$this->items[] = $item;
 			}
+		}
+		else {	//queriedContext == 'By list'
+			$this -> feedName = 'Releases';
+			$list = array();
+
+			$releasesHTML = getSimpleHTMLDOM(self::RELEASES_URL)
+				or returnServerError('Could not connect to releases page.');
```

Comment:
"or returnServerError" after getSimpleHTMLDOM is useless, 'cos getSimpleHTMLDom already raises exception on error. Please remove "or returnServerError" everywhere in this file.